Code/src/main/java/nl/utwente/viskell/ui/components/Lane.java

```diff
@@ -0,0 +1,230 @@
+import nl.utwente.viskell.ui.DragContext;
+
+/**
+ * A single alternative within a ChoiceBlock
+ *
+ */
+public class Lane extends BorderPane implements BlockContainer, ComponentLoader {
+    
+    /** The argument anchors of this alternative */
+    protected List<BinderAnchor> arguments;
+    
+    /** The result anchor of this alternative */
+    protected ResultAnchor result;
+
+    /** Whether the anchor types are fresh*/
+    protected boolean freshAnchorTypes;
+    
+    /** Status of change updating process in this block. */
+    protected boolean firstPhaseInProgress;
+    
+    /** Status of change updating process in this block. */
+    protected boolean finalPhaseInProgress;
+    
+    /** The wrapper to which this alternative belongs */
+    protected ChoiceBlock parent;
+    
+    /** The draggable resizer in the bottom right corner */
+    private Polygon resizer;
+
+    /** The container Node for binder anchors */
+    @FXML protected Pane argumentSpace;
+    
+    /** The resizable area on which blocks can be placed */
+    @FXML protected Pane guardSpace;
+    
+    /** The container Node for result anchors */
+    @FXML protected Pane resultSpace;
+
+    /** A set of blocks that belong to this container */
+    protected Set<Block> attachedBlocks;
+
+    public Lane(ChoiceBlock wrapper) {
+        super();
+        loadFXML("Lane");
+        parent = wrapper;
+        arguments = new ArrayList<>();
+        result = new ResultAnchor(this, wrapper, Optional.empty());
+        attachedBlocks = new HashSet<>();
+        
+        argumentSpace.getChildren().addAll(arguments);
+        resultSpace.getChildren().add(result);
+        setupResizer();
+    }
+
+    /**
+     * Returns the pattern and guard expression for this lane as well as a set of required blocks.
+     * 
+     * Firstly, the expression is generated from the result anchor.
+     * Secondly, the expression is extended with bottom-most blocks within this lane.
+     * Bottom-most blocks are *assumed* to be either deconstructors or expressions resulting in a Bool.
+     * 
+     * @return a pair containing this lane's Alternative and a set of required blocks
+     */
+    public Pair<Case.Alternative,Set<Block>> getAlternative() {
+        Pair<Expression, Set<Block>> pair = result.getLocalExpr();
+        LetExpression guards = new LetExpression(pair.a, true);
+        Set<Block> surroundingBlocks = pair.b;
+        result.extendExprGraph(guards, this, surroundingBlocks);
+        
+        attachedBlocks.stream().filter(Block::isBottomMost).forEach(block -> {
+            if (block instanceof MatchBlock) {
+                guards.addLetBinding(((MatchBlock)block).getPrimaryBinder(), block.getAllInputs().get(0).getFullExpr());
+            }
+            else {
+                block.getAllOutputs().forEach(anchor -> {
+                    guards.addLetBinding(new ConstructorBinder("True", Collections.EMPTY_LIST), anchor.getVariable());
```

Comment:
The last line specifically? It tries to bind the evaluation of a function to True, like a normal guard with a function would do, e.g.

    case e of
      a | someBoolExpression -> b

...except `LetExpression`s don't allow for a binding without a left-hand side, so it becomes `True <- someBoolExpression`.

Does it need explaining in the code?